File: JavaScript/src/Internals/Functions/MathSum/Function_SUMSQ.js
import { Function_N } from '../Function_N.js';
import { Operand } from '../../../Operand.js';
import { FunctionUtil } from '../FunctionUtil.js';

class Function_SUMSQ extends Function_N {
    get Name() {
        return "SumSq";
    }

    constructor(z) {
        super(z);
    }

    evaluate(engine, tempParameter) {
        let args = [];
        for (let i = 0; i < this.z.length; i++) {
            let aa = this.z[i].evaluate(engine, tempParameter);
            if (aa.IsError) {
                return aa;
            }
            args.push(aa);
        }

        let list = [];
        let o = FunctionUtil.F_base_GetList(args, list);
        if (o == false) {
            return this.functionError();
        }

        let d = 0;
        for (let i = 0; i < list.length; i++) {
            let a = list[i];
            d += a * a;
        }
        return Operand.Create(d);
    }
}

export { Function_SUMSQ };

File: java/src/main/java/toolgood/algorithm/internals/functions/mathsum/Function_SUMSQ.java
package toolgood.algorithm.internals.functions.mathsum;

import java.util.ArrayList;
import java.util.List;


import toolgood.algorithm.internals.functions.FunctionBase;
import toolgood.algorithm.internals.functions.Function_N;
import toolgood.algorithm.Operand;
import toolgood.algorithm.internals.functions.FunctionUtil;

public class Function_SUMSQ extends Function_N {
    public Function_SUMSQ(FunctionBase[] funcs) {
        super(funcs);
    }

    @Override
    public Operand Evaluate(Object work, Function<Object, String, Operand> tempParameter) {
        List<Operand> args = new ArrayList<>(funcs.length);
        for (int i = 0; i < funcs.length; i++) {
            Operand aa = funcs[i].Evaluate(work, tempParameter);
            if (aa.IsError()) {
                return aa;
            }
            args.add(aa);
        }

        List<Double> list = new ArrayList<>();
        boolean o = FunctionUtil.F_base_GetList(args, list);
        if (!o) {
            return Operand.Error("Function '{0}' parameter is error!", "SumSQ");
        }

        double d = 0;
        for (int i = 0; i < list.size(); i++) {
            double a = list.get(i);
            d += a * a;
        }
        return Operand.Create(d);
    }

    @Override
    public void toString(StringBuilder stringBuilder, boolean addBrackets) {
        AddFunction(stringBuilder, "SumSQ");
    }
}
